fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/Operator.java

```diff
@@ -86,4 +86,46 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(opType.ordinal(), limit, predicate);
     }
+
+    public abstract static class Builder<O extends Operator, B extends Builder> {
+        protected OperatorType opType;
+        protected long limit = -1;
+        protected ScalarOperator predicate;
```

Comment:
`protected O operator`
Can be reduce more duplicate define code? but this may need set `operator = null` after call `build()` to avoid the builder keep refer to the object